crates/kubelet/src/kubelet.rs

```diff
@@ -63,8 +69,53 @@ impl<T: 'static + Provider + Sync + Send> Kubelet<T> {
             }
         });
 
-        // This informer listens for pod events.
-        let provider = self.provider.clone();
+        // TODO: How should we configure this value? We should eventually have a max pods setting
+        // just like a normal kubelet, so maybe that?
+        let (error_sender, mut error_receiver) = mpsc::channel::<(KubePod, anyhow::Error)>(200);
+        let client_clone = client.clone();
+        let error_handler = tokio::task::spawn(async move {
+            let client = client_clone;
+            while let Some(e) = error_receiver.recv().await {
+                let (pod, err) = e;
+                let json_status = serde_json::json!(
+                    {
+                        "metadata": {
+                            "resourceVersion": "",
+                        },
+                        "status": {
+                            "phase": Phase::Failed,
+                            "message": format!("{}", err),
+                        }
+                    }
+                );
+
+                debug!(
+                    "Setting pod status for {} using {:?}",
+                    pod.name(),
+                    json_status
+                );
+
+                let data = serde_json::to_vec(&json_status).expect("Should always serialize");
+                let pod_resource =
+                    Resource::namespaced::<KubePod>(&pod.namespace().unwrap_or_default());
```

Comment:
Is there simpler way to do this using the `Api` api from kube?

Reply:
Actually yes. I think I was planning on doing something different here and then never did it